plugins/browser-plugin-client-hints/src/index.ts

```diff
@@ -56,6 +56,16 @@ declare global {
 
 let uaClientHints: HttpClientHints;
 
+/**
+ * This function makes sure that the expected array is returned as an array instead of an object.
+ * It handles a problem that in some cases the `navigator.userAgentData.brands` was returned as an object instead of array.
+ */
+function forceArray<T>(array: T[]): T[] {
+  return Object.keys(array).map((e) => {
+    return (array as any)[e];
+  });
+}
+
```

Comment:
Do you have an example of one of the malformed objects? I'll be honest, I'm struggling to imagine a malformed object that passes through this and ends up in the same format as the expected array of objects.

Reply:
Yes, we do have an example, should have put it in the PR description:

```json
{ "0":{"brand":" Not A;Brand","version":"99"}, "1":{"brand":"Chromium","version":"102"}, "2":{"brand":"Google Chrome","version":"102"} }
```

It was [reported here](https://github.com/snowplow/snowplow-javascript-tracker/issues/1082#issuecomment-1162392887).

Reply:
I should have looked at the issue 🤦

Reply:
Looks good. If you want to make TypeScript happy, you could do something like this:

```
function forceArray<T>(array: T[] | Record<string, T>): T[] {
  if (Array.isArray(array)) return array;

  return Object.keys(array).map((e) => {
    return array[e];
  });
}
```

Reply:
Looks better, thanks!